docs/root/intro/life_of_a_request.rst

```diff
@@ -0,0 +1,181 @@
+.. _life_of_a_request:
+
+Life of a Request
+=================
+
+Below we follow the events that take place after the arrival of a request at the Envoy proxy from a
+downstream client, the dispatch to an upstream backend and the corresponding response path.
+
+Network topology
+----------------
+
+Envoy can be used in a wide variety of networking topologies. We focus on the inner operation
+of Envoy below, but briefly we address how Envoy relates to the rest of the network in this section.
+
+Envoy originated as a `service mesh
+<https://blog.envoyproxy.io/service-mesh-data-plane-vs-control-plane-2774e720f7fc>`_ sidecar proxy,
+factoring out load balancing, routing, observability, security and discovery services from
+applications. In the service mesh model:
+
+* Ingress listeners take requests from other nodes in the service mesh and forward them to the
+  local application. Responses from the local application flow back through Envoy to the other
+  nodes in the service mesh.
+* Egress listeners take requests from the local application and forward them to other nodes in the
+  network. These nodes will also be typically running Envoy and accepting the request via their
+  ingress listeners.
+
+.. note::
+  TODO: add a service mesh diagram
+
+Envoy is used in a variety of domains beyond the service mesh. For example, it can also act
+as an internal load balancer and as an edge networking ingress/egress proxy.
+
+.. note::
+  TODO: add topology diagrams for edge and ILB.
+
+Envoy may be configured in multi-tier topologies for scalability and reliability:
+
+.. note::
+  TODO: add topology diagrams for multi-tier.
+
+In all the above cases, a request will arrive at a specific Envoy via TCP, UDP or Unix Domain
+Sockets from a downstream. Envoy will forward requests to upstreams via TCP or Unix Domain
+Sockets. We focus on a single Envoy proxy that is part of one of these network topologies
+below.
+
+Configuration
+-------------
+
+Envoy is a platform with a high degreee of extensibility. This results in a combinatorial
+explosion of possible request paths, depending on:
+
+* L3/4 protocol, e.g. TCP, UDP, Unix Domain Sockets.
+* L7 protocol, e.g. HTTP/1, HTTP/2, HTTP/3, gRPC, Thrift, Dubbo, Kafka, Redis and various databases.
+* Transport socket, e.g. plain text, TLS, ALTS.
+* Routing, e.g. PROXY protocol, original destination, dynamic forwarding.
+* Authentication and authorization.
+* Circuit breakers and outlier detection.
+* Many other networking, HTTP, listener, access logging, health checking, tracing and stats
+  extensions and configuration options.
+
+It's helpful to limit the nature of the configuration and request to provide an illustrative
+narrative below, so we opt for the following parameters:
```

Comment:
This is something I 100% suck at, but it may be worth doing a pass pretending you're ESL (or asking someone who is to do so) and using more common words where you can.

Reply:
I also am terrible at this. Please correct places that things seem that they aren't ESL friendly. Hopefully we can get a final pass and clean these up at the end.